taskvine/src/manager/taskvine.h

```diff
@@ -48,6 +48,11 @@ typedef enum {
 	VINE_SUCCESS_ONLY = 8, /**< Only return this output file if the task succeeded. */
 } vine_file_flags_t;
 
+typedef enum {
+	VINE_SHAREOK = 0, /**< Schedule this file to be shared between peers where available. (default) **/
+	VINE_NOSHARE = 1  /**< Do not schedule this file to be shared between peers. **/
+} vine_file_transfer_flags_t;
```

Comment:
vine_file_share_peer_mode_t ?   

My hunch is that flags is a little to generic, and the options are specifically for peers, not transfers in general.